internal/infrastructure/storage/db/pg/migration/20230504102232_init.up.sql

```diff
@@ -0,0 +1,63 @@
+CREATE TABLE market (
+    name VARCHAR(50) NOT NULL PRIMARY KEY,
+    base_asset VARCHAR(255) NOT NULL,
+    quote_asset VARCHAR(255) NOT NULL,
+    base_asset_precision INTEGER NOT NULL DEFAULT 8,
+    quote_asset_precision INTEGER NOT NULL DEFAULT 8,
+    tradeable BOOLEAN NOT NULL DEFAULT FALSE,
+    strategy_type INTEGER NOT NULL DEFAULT 0,
+    base_price INTEGER NOT NULL DEFAULT 0,
+    quote_price INTEGER NOT NULL DEFAULT 0,
+    UNIQUE (base_asset, quote_asset)
+);
+
+CREATE TABLE fee (
+    id SERIAL PRIMARY KEY,
+    base_asset_fee BIGINT NOT NULL,
+    quote_asset_fee BIGINT NOT NULL,
+    type VARCHAR(10) NOT NULL CHECK (type IN ('percentage', 'fixed')),
+    fk_market_name VARCHAR(50) NOT NULL,
+    FOREIGN KEY (fk_market_name) REFERENCES market (name)
+);
```

Comment:
Maybe better to store base/quote asset fee directly in the market table as fields `percentage_base_fee | percentage_quote_fee | fixed_base_fee | fixed_quote_fee`?

Reply:
since we basically have same fields that differs in type(fixed, percentage) i thought it is nicer to have it in separate table which is in 1:M relation to the market table.

Reply:
is there a way to enforce that every record in market table must have fixed and percentage fee in this fee table?

Reply:
during the review, I saw the same thing done for swaps, no need to change this.